frontend/js/src/user/playlists/components/ImportSoundCloudPlaylistModal.tsx

```diff
@@ -0,0 +1,203 @@
+import React from "react";
+import NiceModal, { useModal } from "@ebay/nice-modal-react";
+import { toast } from "react-toastify";
+import { Link } from "react-router-dom";
+import GlobalAppContext from "../../../utils/GlobalAppContext";
+import { ToastMsg } from "../../../notifications/Notifications";
+import Loader from "../../../components/Loader";
+
+type ImportPLaylistModalProps = {
+  listenlist: JSPFTrack[];
+};
+
+export default NiceModal.create((props: ImportPLaylistModalProps) => {
+  const modal = useModal();
+
+  const closeModal = React.useCallback(() => {
+    modal.hide();
+    document?.body?.classList?.remove("modal-open");
+    setTimeout(modal.remove, 200);
+  }, [modal]);
+
+  const { APIService, currentUser } = React.useContext(GlobalAppContext);
+  const [playlists, setPlaylists] = React.useState<SoundCloudPlaylistObject[]>(
+    []
+  );
+  const [newPlaylists, setNewPlaylists] = React.useState<JSPFPlaylist[]>([]);
+  const [playlistLoading, setPlaylistLoading] = React.useState<string | null>(
+    null
+  );
+
+  React.useEffect(() => {
+    async function getUserSoundcloudPlaylists() {
+      try {
+        const response = await APIService.importPlaylistFromSoundCloud(
+          currentUser?.auth_token
+        );
+
+        if (!response) {
+          return;
+        }
+
+        setPlaylists(response);
+      } catch (error) {
+        toast.error(
+          <ToastMsg
+            title="Error loading playlists"
+            message={
+              error?.message === "Forbidden" ? (
+                <>
+                  Session has expired. Please reconnect to{" "}
+                  <Link to="/settings/music-services/details/">Soundcloud</Link>
+                  .
+                </>
+              ) : (
+                error?.message ?? error
+              )
```

Comment:
Instead of handling error message `Forbidden`, you should handle the error `Unauthorized` and `Not Found`. [link](https://developers.soundcloud.com/docs/api/explorer/open-api#/me/get_me_playlists)